src/rules/prefer-to-have-value.js

```diff
@@ -21,62 +24,117 @@ export const meta = {
 };
 const messageId = "use-to-have-value";
 
-export const create = (context) => ({
-  // expect(element.value).toBe('foo') / toEqual / toStrictEqual
-  [`CallExpression[callee.property.name=/to(Be|(Strict)?Equal)$/][callee.object.arguments.0.property.name=value][callee.object.callee.name=expect]`](
-    node
-  ) {
-    context.report({
-      messageId,
-      node,
-      fix(fixer) {
+export const create = (context) => {
+  function validateQueryNode(nodeWithValueProp) {
+    const queryNode =
+      nodeWithValueProp.type === "Identifier"
+        ? getAssignmentForIdentifier(context, nodeWithValueProp.name)
+        : getInnerNodeFrom(nodeWithValueProp);
+
+    if (!queryNode || !queryNode.callee) {
+      return {
+        isValidQuery: false,
+        isValidElement: false,
+      };
+    }
+
+    const query = queryNode.callee.name || queryNode.callee.property.name;
+    const queryArg = queryNode.arguments[0] && queryNode.arguments[0].value;
+    const isValidQuery = queries.includes(query);
+    const isValidElement =
+      query.match(/^(get|find|query)ByRole$/) &&
+      ["textbox", "dropdown"].includes(queryArg);
+    return { isValidQuery, isValidElement };
+  }
+  return {
+    // expect(element.value).toBe('foo') / toEqual / toStrictEqual
+    // expect(<query>.value).toBe('foo') / toEqual / toStrictEqual
+    // expect((await <query>).value).toBe('foo') / toEqual / toStrictEqual
+    [`CallExpression[callee.property.name=/to(Be|(Strict)?Equal)$/][callee.object.arguments.0.property.name=value][callee.object.callee.name=expect]`](
+      node
+    ) {
+      const { isValidQuery, isValidElement } = validateQueryNode(
+        node.callee.object.arguments[0].object
+      );
+      function fix(fixer) {
         return [
           fixer.removeRange([
-            node.callee.object.arguments[0].object.range[1],
+            node.callee.object.arguments[0].property.range[0] - 1,
```

Comment:
what is the `-1` for?  generally hardcoded values in fixers can be problematic, unless you're inserting an arg (which i'd love to find a better way).

Reply:
If the object is wrapped in parentheses (like an inline await statement `expect((await screen.findByRole('textbox')).value)`), removing from the end of its range to the end of the property's range also gets rid of the closing wrapping parenthesis `expect((await screen.findByRole('textbox'))`. It does feel kinda icky to use magical numbers but I couldn't find another way to get around that than to just assume that the `.` preceding `value` will be just one over in the vast majority of cases. Definitely open to other suggestions though.

Reply:
@juzerzarif, i did some digging and i think I found a winner!  you can use `context.getSourceCode().getTokenBefore(node)` or `getTokenAfter()`.  I discovered this by changing `.value` to be `.  value` which broke both the test for this use case as well as the ones i was using in to-have-class.  By switching to these functions they're now completely immune to spacing issues. :)  I hope you don't mind I just pushed it to your branch along w/ some other changes just extracting some variables out for readability.